reco_utils/evaluation/python_evaluation.py

```diff
@@ -298,6 +299,44 @@ def auc(
     return roc_auc_score(y_true, y_pred)
 
 
+def binarize_datasets(
+    df_true,
+    df_pred,
+    threshold,
+    col_rating=DEFAULT_RATING_COL,
+    col_prediction=DEFAULT_PREDICTION_COL,
+):
+    """
+    Transform the ground truth and prediction datasets in the following way:
+      - Ground truth's ratings become binary (1 if the rating exceed the threshold and 0 if it doesn't)
+      - Prediction's ratings are scaled down to 0..1 range which effectively give us the probabilities for
+        accurate/high rating.
+    This is used to prepare the datasets for computing logloss (see below).
+
+    Args:
+        df_true (pd.DataFrame): True data
+        df_pred (pd.DataFrame): Predicted data
+        threshold (float): Positivity threshold
+        col_rating (str): column name for rating
+        col_prediction (str): column name for prediction
+
+    Returns:
+        pd.DataFrame, pd.DataFrame: The binary ground truth dataset and and the scaled prediction dataset.
+    """
+
+    df_true_bin = df_true.copy()
+    df_true_bin[col_rating] = df_true_bin[col_rating].apply(
+        lambda x: 1 if x >= threshold else 0
+    )
```

Comment:
if we use numpy it is around x100 faster:
```
threshold = 3
lower = 0
upper = 1
 np.where(df[col_rating].values > threshold, upper, lower)
```

also, can we please create an option with `inplace` so we don't have to copy the df?

Reply:
Used the `binarize` function from the other PR.